src/NodaTime/Duration.cs

```diff
@@ -1154,13 +1177,18 @@ void IXmlSerializable.WriteXml([NotNull] XmlWriter writer)
         [Pure]
         private long ToInt64NanosecondsUnchecked() => unchecked(days * NanosecondsPerDay + nanoOfDay);
 
+        // NOTE: Do not use from internal code. Use decimal instead. TODO: Add an attribute, and use a Roslyn analyzer to validate.
```

Comment:
It would be good to disallow that within Roslyn, yes. But we should *also* disallow using BigInteger in almost every other situation. Basically this is a bit of a tricky situation.